source/wp-content/themes/wporg-developer-2023/functions.php

```diff
@@ -352,3 +377,25 @@ function rename_comments_meta_box( $post_type, $post ) {
 function update_prism_css_path( $path ) {
 	return '/stylesheets/prism.css';
 }
+
+/**
+ * Filters breadcrumb items for the site-breadcrumb block.
+ *
+ * @return array
+ */
+function set_site_breadcrumbs() {
+	$breadcrumbs = array();
+
+	foreach ( get_breadcrumbs()->items as $crumb ) {
+		// Get the link and title from the breadcrumb.
+		// 4 capture groups to handle variation in the markup.
+		// This code assumes that an <a> tag or a string is returned from the breadcrumb.
+		preg_match( '/<a(.*?)href="([^"]+)"(.*?)>(.*?)<\/a>/', $crumb, $matches );
+		$breadcrumbs[] = array(
+			'url' => isset( $matches[2] ) ? $matches[2] : '',
+			'title' => isset( $matches[4] ) ? $matches[4] : $crumb,
+		);
```

Comment:
```suggestion
		preg_match( '!<a[^>]+href="(?P<href>[^"]+)"[^>]+>(?P<title>.+)</a>!', $crumb, $matches );
		$breadcrumbs[] = array(
			'url'   => $marches['href'] ?? '',
			'title' => $matches['title'] ?? $crumb,
		);
```

Using a non-/ delimiter makes it easier, I would normally use `/`, `!`, `#` or `~` depending upon what characters are in the regex.. if `/` then `!`, if using `!` within negative lookahead/behinds then `#` or `~`.

Using named groups helps readability, and since 1 & 3 aren't being used there's no point in capturing these groups either.

Using the PHP 7 Null-Coalesce makes it a bit nicer to read as well.

Reply:
Nice, that's definitely cleaner. I had to tweak the regex a bit:

`'!<a[^>]+href="(?P<href>[^"]+)"[^>]+>(?P<title>.+)</a>!'`
to:
`!<a[^>]+href="(?P<href>[^"]+)"[^>]*>(?P<title>.+)</a>!`